nautobot/extras/group_sync.py

```diff
@@ -0,0 +1,42 @@
+"""Additional functions to process an OAuth2/OIDC user."""
+
+import logging
+
+from django.conf import settings
+from django.contrib.auth.models import Group
+
+logger = logging.getLogger(__name__)
+
+
+CLAIMS_GROUP_NAME = getattr(settings, "NAUTOBOT_SSO_CLAIMS_GROUP", None)
```

Comment:
1. It would be good to define these settings in `nautobot/core/settings.py` and document them in `nautobot/core/settings.yaml`.

2. Would it be desirable or make sense to support definition of these configs via environment variables?

3. Should CLAIMS_GROUP_NAME just default to `"groups"` if not otherwise set, as described below?

Reply:
Well we're on the same page. I've honestly wanted to make the settings overall easier for people to provide the data without having to override the `nautobot_config.py`. Locally I was going to do env variables but it definitely seems like you're gonna hit a wall. Especially with things that aren't neatly strings (e.g. lists / objects). I was actually asking in Slack the other day how people would feel about the ability to define a YAML file that got loaded.

Reply:
I'm not sure that defining a YAML file is much easier than defining a Python file, but environment variables for strings and lists goes pretty far in most cases :-) Appreciate your thinking about things!

Reply:
Specifically my pain point is configuration data for plugins.

Reply:
Very fair point.